Sources/SwiftSyntax/RawSyntax.swift

```diff
@@ -172,6 +174,13 @@ extension RawSyntax {
     kind.isUnknown
   }
 
+  var hasError: Bool {
+    switch rawData.payload {
+    case .materializedToken(_): return self.presence == .missing
```

Comment:
I would prefer to detect the presence of the node purely based on the `materializedToken` but I’m worried that will break if we change whether tokens without source text but with trivia are considered missing.